module-services/service-desktop/WorkerDesktop.cpp

```diff
@@ -111,97 +113,128 @@ void WorkerDesktop::reset()
     }
 }
 
-bool WorkerDesktop::handleMessage(uint32_t queueID)
+bool WorkerDesktop::handleMessage(std::uint32_t queueID)
 {
+    bool result       = false;
     auto &queue       = queues[queueID];
     const auto &qname = queue->GetQueueName();
 
     if (qname == sdesktop::RECEIVE_QUEUE_BUFFER_NAME) {
-        std::string *receivedMsg = nullptr;
-        if (!queue->Dequeue(&receivedMsg, 0)) {
-            LOG_ERROR("handleMessage failed to receive from \"%s\"", sdesktop::RECEIVE_QUEUE_BUFFER_NAME);
-            return false;
-        }
+        result = handleReceiveQueueMessage(queue);
+    }
+    else if (qname == sdesktop::SEND_QUEUE_BUFFER_NAME) {
+        result = handleSendQueueMessage(queue);
+    }
+    else if (qname == SERVICE_QUEUE_NAME) {
+        result = handleServiceQueueMessage(queue);
+    }
+    else if (qname == sdesktop::IRQ_QUEUE_BUFFER_NAME) {
+        result = handleIrqQueueMessage(queue);
+    }
+    else {
+        LOG_INFO("handeMessage got message on an unhandled queue");
+    }
 
-        using namespace sdesktop::endpoints;
-        auto factory = EndpointFactory::create(securityModel.getEndpointSecurity());
-        auto handler = std::make_unique<MessageHandler>(ownerService, messageProcessedCallback, std::move(factory));
+    return result;
+}
+
+bool WorkerDesktop::handleReceiveQueueMessage(std::shared_ptr<sys::WorkerQueue> &queue)
+{
+    if (!initialized) {
+        return false;
+    }
+    std::string *receivedMsg = nullptr;
+    if (!queue->Dequeue(&receivedMsg, 0)) {
+        LOG_ERROR("handleMessage failed to receive from \"%s\"", sdesktop::RECEIVE_QUEUE_BUFFER_NAME);
+        return false;
+    }
 
-        parser.setMessageHandler(std::move(handler));
-        parser.processMessage(std::move(*receivedMsg));
+    using namespace sdesktop::endpoints;
+    auto factory = EndpointFactory::create(securityModel.getEndpointSecurity());
+    auto handler = std::make_unique<MessageHandler>(ownerService, messageProcessedCallback, std::move(factory));
 
-        delete receivedMsg;
+    parser.setMessageHandler(std::move(handler));
+    parser.processMessage(std::move(*receivedMsg));
+
+    delete receivedMsg;
+    return true;
+}
+bool WorkerDesktop::handleSendQueueMessage(std::shared_ptr<sys::WorkerQueue> &queue)
+{
+    if (!initialized) {
+        return false;
+    }
+    std::string *sendMsg = nullptr;
+    if (!queue->Dequeue(&sendMsg, 0)) {
+        LOG_ERROR("handleMessage xQueueReceive failed for %s.", sdesktop::SEND_QUEUE_BUFFER_NAME);
+        return false;
     }
-    else if (qname == sdesktop::SEND_QUEUE_BUFFER_NAME) {
-        std::string *sendMsg = nullptr;
-        if (!queue->Dequeue(&sendMsg, 0)) {
-            LOG_ERROR("handleMessage xQueueReceive failed for %s.", sdesktop::SEND_QUEUE_BUFFER_NAME);
-            return false;
-        }
 
-        bsp::usbCDCSend(sendMsg);
-        delete sendMsg;
+    bsp::usbCDCSend(std::move(sendMsg));
```

Comment:
After applying my suggestion this will probably not compile as `bsp::usbCDCSend` takes a pointer. It would require to refactor `bsp::usbCDCSend` to take r-val reference. So maybe we should leave it as it was. ;)

Reply:
So I think that `std::move` should be removed because it further suggest that this is an actual message object and this makes things even worse.

Reply:
Done